glacier_mapping/models/frame.py

```diff
@@ -98,6 +105,35 @@ def infer(self, x):
         with torch.no_grad():
             return self.model(x).permute(0, 2, 3, 1)
 
+    def predict(self, y_hat):
```

Comment:
Modularizing like this is much better. Though, can you change the name from `predict` to a more appropriate verb? Predict sounds like you give x as input and it returns y_hat. This is taking a probability map and outputting a segmentation. Maybe something like `self.threshold` or `self.segment` is more appropriate?